sliceencoder.go

```diff
@@ -84,8 +95,8 @@ var (
 	zero = uintptr(0)
 )
 
-func (e *SliceEncoder) sliceInstr() {
-	enc := NewSliceEncoder(reflect.New(e.tt.Elem()).Elem().Interface())
+func (e *SliceEncoder) sliceInstr(cfg Config) {
```

Comment:
It would increase the size of each encoder by ~1 byte for configuration that is functionless after encoder initialisation, it may be more appropriate to put the config on the encoder if instruction behavior was more dynamic i.e. toggling sorting of keys without regenerating encoders.

Reply:
I think you could make the same argument for a few fields on those structs, however it would lead to some quite unwieldy function signatures were we to take the same approach. Especially given the functions benefit from the pointer receiver anyway. 

It feels like we can afford the extra width of a field in this instance (non-exported).

Reply:
Updated - cfg has been relocated the encoder.